spinalcordtoolbox/centerline/core.py

```diff
@@ -237,9 +241,30 @@ def get_centerline(im_seg, param=ParamCenterline(), verbose=1, remove_temp_files
                      else "centerline.nii.gz")
         im_centerline.save(os.path.join(tmp_folder, fname_ctr), mutable=True)
 
+    # If 'phys' is specified, adjust centerline coordinates (`Centerline.points`) and
+    # derivatives (`Centerline.derivatives`) to physical ("phys") space and native (`im_seg`) orientation.
+    if space == 'phys':
+        # Transform centerline to physical coordinate system
+        arr_ctl = im_seg.transfo_pix2phys([[x_centerline_fit[i], y_centerline_fit[i], z_ref[i]]
+                                           for i in range(len(x_centerline_fit))]).T
+        # Adjust derivatives with pixel size
+        nx, ny, nz, nt, px, py, pz, pt = im_seg.change_orientation(native_orientation).dim
+        arr_ctl_der = np.array([x_centerline_deriv * px, y_centerline_deriv * py, np.ones_like(z_ref) * pz])
+    else:
+        arr_ctl = np.array([x_centerline_fit, y_centerline_fit, z_ref])
+        arr_ctl_der = np.array([x_centerline_deriv, y_centerline_deriv, np.ones_like(z_ref)])
+
+    centerline = Centerline(
+        points_x=arr_ctl[0].tolist(),
+        points_y=arr_ctl[1].tolist(),
+        points_z=arr_ctl[2].tolist(),
+        deriv_x=arr_ctl_der[0].tolist(),
+        deriv_y=arr_ctl_der[1].tolist(),
+        deriv_z=arr_ctl_der[2].tolist()
+    )
```

Comment:
Ahhh, this is a very good point. I like this idea a lot!!

Essentially, we would make it so that wherever `straightening._get_centerline()` used to be called, we call `get_centerline(space='phys')` instead, then feed the resulting `arr_ctl` and `arr_ctl_der` to `Centerline` at the site of the caller, rather than within the function itself.

In other words:

* We keep the thrust of commit https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4010/commits/784704c4a15f24a3f9cd54c2d49cca976cf88f69
* But we reverse all of the fiddly changes to https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4010/commits/415db4d1a0407d3fdda00b7932b72970441aff99, since these calls were just fine working in pixel space without the `Centerline` objects.

Reply:
The only downside is that this would hinder efforts to encourage developers to use `Centerline` objects, since `get_centerline` wouldn't return them. So, the onus is on the caller to know that `Centerline` exists, and when to import it and use it.

But, maybe this is a good thing, given your observation that `Centerline` may not be compatible with pixel-space arrays.

Reply:
I've reverted the commits that modify `get_centerline`, preserving the existing function calls. Then, I redid the commit that removes `_get_centerline`.

You're absolutely correct -- this drastically reduces the scope of this PR, and only creates `Centerline` objects for `space="phys"` specifically, which avoids the concerns described in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4010#discussion_r1107400160.